### PR TITLE
fix(otlp-trace): isolate Signal B by session

### PR DESCRIPTION
**Pull request by fangxiu-wf**
## Summary

- persist session identity on each OTLP turn buffer and learn it from later records when necessary
- allow Signal B preemption only when both turns have known, matching `gen_ai.session.id` values
- keep different-session and unknown-session buffers isolated until their own terminal, idle timeout, buffer cap, or shutdown signal
- add Codex A/B/A interleaving and known/missing session regressions so each turn converts exactly once

This is a focused follow-up to #115. Signal A terminal handling, transcript extraction, and synthetic `other` events are unchanged.

## Verification

- targeted Signal B/grouping tests — 25 passed
- all flusher tests — 134 passed
- `npm run typecheck` — passed
- replayed 7/28 Codex JSONL: 7,623 records, 0 missing session IDs, 77 terminal events, 0 confirmed-same-session preemptions, 0 buffers left open

### CODE REVIEW

**Review by ralf0131 — APPROVED** [review]
Changes look correct and well-tested. The behavioral change for missing-session records (now always preempts, matching legacy behavior) is intentional and documented.

---
*Automated review by github-manager-bot*

**Comment by rangemer333-cell** [comment]
结论：PR #207 能解决注释中的 Codex 并发 Session 问题。对于 Windows Codex 当前输出格式，没有阻塞性问题，可以合并。
但有一个非 Codex 范围的 P2 风险。
Review finding
[P2] 任意一侧缺少 gen_ai.session.id 时仍会无条件触发 Signal B。
当前判断是：
if (incomingSessionId && buf.sessionId && incomingSessionId !== buf.sessionId) {
  continue;
}

buf.completed = true;
this.triggerFlush(buf, false);
因此以下场景仍会提前拆分：
B buffer 已知 session=B
A 的首条记录暂时没有 session.id
→ 条件无法确认不同 Session
→ 提前 flush B
→ B 后续记录重建 buffer
→ 仍可能生成两套 ENTRY/AGENT
PR 还删除了 main 原有的 hasLlmResponse 保护，明确恢复“缺失 session 时按旧逻辑抢占”。这不影响当前 Codex，因为 Codex builder 给所有事件都带了 gen_ai.session.id，但对字段不完整的其他 Agent 是一个回退风险。[相关代码](https://github.com/alibaba/loongsuite-pilot/blob/51c4d202c8d71af156285eb7d0a15099067002ec/src/flushers/otlp-trace-flusher.ts#L228-L239)
建议后续改成：
两边 session 都存在且相同：允许 Signal B。
两边 session 都存在且不同：允许并发。
任意一边未知：不要基于 Signal B 提前导出，依赖 terminal、idle timeout、buffer cap 或 shutdown。
增加“已知 session + 缺失 session”两个方向的交错测试。

**Comment by fangxiu-wf** [comment]
感谢指出，这个 P2 场景成立，已在 5e9033c 修复：Signal B 现在只有在两侧 session 均存在且明确相同时才允许 preempt；session 不同或任意一侧缺失都会保留 buffer，等待自身 terminal / idle timeout / cap / shutdown。已增加“已知 buffer + 未知 incoming”“未知 buffer + 已知 incoming”以及双方未知的回归测试，同时保留 same-session、different-session 和 Codex A/B/A 覆盖。定向测试 25/25、全部 flusher 测试 134/134、typecheck 均通过。

**Review by rangemer333-cell — APPROVED** [review]

File: src/flushers/otlp-trace-flusher.ts [hunk 1]
```diff
@@ -275,11 +259,14 @@ export class OtlpTraceFlusher extends BaseFlusher {
         keySource: source,
         keyValue: value,
         agentType,
+        sessionId: incomingSessionId,
         records: [],
         completed: false,
```
Comment (ralf0131) [inline src/flushers/otlp-trace-flusher.ts:264]: Good improvement — storing `sessionId` on the buffer and learning it from later records (line 268-269) eliminates the previous ambiguity where missing session info triggered the `hasLlmResponse` fallback. The new rule (only preempt when both sessions are known and match) is strictly safer.

**Review by ralf0131 — APPROVED** [review]
### Summary
This PR simplifies the Signal B turn-boundary logic by requiring both the incoming and buffered turns to have known, matching session IDs before preempting. This eliminates the previous  fallback that could cause incorrect preemptions when session info was missing.

The change is well-motivated, the code is cleaner, and the 4 new test cases thoroughly cover the edge cases (interleaved sessions, missing session IDs, late session ID learning). LGTM.

---
*Automated review by github-manager-bot*

**Comment by ralf0131** [comment]
📋 **Issue Evaluation**

Category: `bug` | Status: **Confirmed & Fix Merged**

The OTLP trace Signal B cross-session preemption issue is valid. The fix correctly isolates buffers by session ID and only allows preemption when sessions match.

PR #207 merged into integration branch. Awaiting merge to main.
